host/CMakeLists.txt

```
@@ -1,20 +1,38 @@
-cmake_minimum_required(VERSION 3.11)
-
-# Compile for x86_64 on Mac as we can't support the M1 ARM architecture yet
-set(CMAKE_OSX_ARCHITECTURES "x86_64" CACHE INTERNAL "")
+cmake_minimum_required(VERSION 3.21)
+project(xscope_fileio)
 
+# Set build type to Release if not set
 set(CMAKE_BUILD_TYPE "Release")
 
-project(xscope_fileio)
+# Universal Mac OS binary
+set(CMAKE_OSX_ARCHITECTURES arm64 x86_64)
 
+# Add executable
 add_executable(xscope_host_endpoint xscope_io_host.c)
 
-set_property(TARGET xscope_host_endpoint PROPERTY C_STANDARD 99)
-
-target_include_directories(xscope_host_endpoint PRIVATE ../xscope_fileio
-                                                        $ENV{XMOS_TOOL_PATH}/include)
-
-find_library(XSCOPE_ENDPOINT_LIB NAMES xscope_endpoint.so xscope_endpoint.lib
-                                 PATHS $ENV{XMOS_TOOL_PATH}/lib)
-
-target_link_libraries(xscope_host_endpoint PRIVATE ${XSCOPE_ENDPOINT_LIB})
+# Include directories
+target_include_directories(
+  xscope_host_endpoint 
+  PRIVATE ../xscope_fileio $ENV{XMOS_TOOL_PATH}/include
+)
+
+# Set the output directory to the same directory as the CMake file
+set_target_properties(xscope_host_endpoint PROPERTIES
+  C_STANDARD 99
+  RUNTIME_OUTPUT_DIRECTORY "${CMAKE_SOURCE_DIR}"
+)
+
+# Find the xscope endpoint library
+find_library(
+  XSCOPE_ENDPOINT_LIB
+  NAMES xscope_endpoint.so xscope_endpoint.lib
+  PATHS $ENV{XMOS_TOOL_PATH}/lib
+)
+
+# Check if the library was found and link it
+if (XSCOPE_ENDPOINT_LIB)
+  target_link_libraries(xscope_host_endpoint PRIVATE ${XSCOPE_ENDPOINT_LIB})
+  message(STATUS "Found xscope endpoint library: ${XSCOPE_ENDPOINT_LIB}")
+else ()
+  message(FATAL_ERROR "xscope endpoint library not found in XMOS_TOOL_PATH/lib.")
```

```suggestion
  message(FATAL_ERROR "xscope endpoint library not found in $ENV{XMOS_TOOL_PATH}/lib.")
```
if you want to print the actual path it looked in